### PR TITLE
Support isl_bool pointer arguments

### PR DESCRIPTION
**Pull request by iasakura**
Hi! Thank you for very useful library!
I recently found that some functions taking a `isl_bool *` as an argument (e.g., `isl_pw_qpolynomial_bound`) are not supported by islpy. This is because of a recent update of isl 0.20 which changes `int*` argument to `isl_bool*` arguments.
The following is an excerpt from the release note.
>  Several functions returning an extra boolean value through an int * argument
now do so through an isl_bool * argument. The returned values are the same,
only the type of the pointer has been changed

This PR is update for that change. Because there seems to be no functions which takes `int*` arguments in ISL any more, I simply changed the case for `int*` to `isl_bool*`.

### CODE REVIEW

File: gen_wrap.py [hunk 1]
```diff
@@ -1368,14 +1368,18 @@ def emit_context_check(arg_idx, arg_name):
 
             docs.append(":param %s: string" % arg.name)
 
-        elif arg.base_type == "int" and arg.ptr == "*":
+        elif arg.base_type == "isl_bool" and arg.ptr == "*":
```
Comment (inducer) [inline gen_wrap.py]: Please add an assertion that supports your claim that no more `int *` arguments are present.

File: gen_wrap.py [hunk 2]
```diff
@@ -1368,14 +1368,18 @@ def emit_context_check(arg_idx, arg_name):
...
-                ret_vals.append("{c_name}[0]".format(c_name=c_name))
-                ret_descrs.append("%s (integer)" % arg.name)
+                ret_vals.append("({c_name}[0] == lib.isl_bool_true)".format(c_name=c_name))
+                ret_descrs.append("%s (bool)" % arg.name)
+                check('if {c_name}[0] == lib.isl_bool_error:'.format(c_name=c_name))
+                with Indentation(check):
+                    check('raise Error("call to \\"{0}\\" failed: %s" '
+                            '% _get_last_error_str(_ctx_data))'.format(meth.c_name))
             else:
                 raise SignatureNotSupported("int *")
```
Comment (inducer) [inline gen_wrap.py]: Please also update this error message.

File: gen_wrap.py [hunk 3]
```diff
@@ -1368,14 +1368,18 @@ def emit_context_check(arg_idx, arg_name):
...
 
                 passed_args.append(c_name)
-                ret_vals.append("{c_name}[0]".format(c_name=c_name))
-                ret_descrs.append("%s (integer)" % arg.name)
+                ret_vals.append("({c_name}[0] == lib.isl_bool_true)".format(c_name=c_name))
+                ret_descrs.append("%s (bool)" % arg.name)
+                check('if {c_name}[0] == lib.isl_bool_error:'.format(c_name=c_name))
+                with Indentation(check):
+                    check('raise Error("call to \\"{0}\\" failed: %s" '
+                            '% _get_last_error_str(_ctx_data))'.format(meth.c_name))
```
Comment (inducer) [inline gen_wrap.py]: If possible, could you add a test to demonstrate that this operates as intended?

File: gen_wrap.py (hunk 1 above)
Comment (iasakura) [reply gen_wrap.py]: I changed it to support both isl_bool and int cases. I think it is better in terms of backward compatibility.

File: gen_wrap.py (hunk 2 above)
Comment (iasakura) [reply gen_wrap.py]: Updated.

File: gen_wrap.py (hunk 3 above)
Comment (iasakura) [reply gen_wrap.py]: I added a simple test case which calls PwQPolynomial.bound, which was not supported in the previous version.

**Comment by iasakura** [comment]
Sorry for the too late reply. I fixed code and merged master. Could you please check the changes?

**Comment by inducer** [comment]
Thanks for your contribution!

**Pull request merged**